app/services/oidc_jwt_token_validate.rb

```diff
@@ -0,0 +1,56 @@
+class OidcJwtTokenValidate
+  attr_reader :decoded_token
+  delegate :logger, to: :Rails
+
+  def initialize(jwt_token)
+    @jwt_token = jwt_token
+  end
+
+  def valid_jwt?
+    valid_signature? && type_bearer? && valid_aud?
+  rescue JWT::InvalidAudError
+    logger.error "OidcJwtToken SSO: Invalid token."
+    false
+  rescue JWT::ImmatureSignature
+    logger.error "OidcJwtToken SSO: JWT is for the future."
+    false
+  rescue JWT::ExpiredSignature
+    logger.error "OidcJwtToken SSO: Signature has expired."
+    false
+  rescue JWT::DecodeError
+    logger.error "OidcJwtToken SSO: Failed to decode JWT."
+    false
+  end
+
+  private
+
+  def generate_jwks_uri
+    response = RestClient::Request.execute(
+      :url => Setting['oidc_jwks_url'],
+      :method => :get,
+      :verify_ssl => true
+    )
+    rsa_public_json = JSON.parse(response)['keys'].first
```

Comment:
Why do we assume, that there is only one key? Can't we pass the whole keystore to `JWT.decode`? The JWT contains the key id, so it should be able to find the correct one.